plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/S3SelectRecordCursorProvider.java

```diff
@@ -109,4 +122,64 @@ public Optional<ReaderRecordCursorWithProjections> createRecordCursor(
             return Optional.empty();
         }
     }
+
+    private boolean hasFilters(
+            Properties schema,
+            TupleDomain<HiveColumnHandle> effectivePredicate,
+            List<HiveColumnHandle> readerColumns)
+    {
+        //There are no effective predicates and readercolumns and columntypes are identical to schema
+        //means getting all data out of S3. We can use S3 GetObject instead of S3 SelectObjectContent in these cases.
+        if (effectivePredicate.isAll()) {
+            return !isEquivalentSchema(readerColumns, schema);
+        }
+        return true;
+    }
+
+    private boolean isEquivalentSchema(List<HiveColumnHandle> readerColumns, Properties schema)
+    {
+        Set<String> projectedColumnNames = getColumnProperty(readerColumns, HiveColumnHandle::getName);
+        Set<String> projectedColumnTypes = getColumnProperty(readerColumns, column -> column.getHiveType().getTypeInfo().getTypeName());
+        return isEquivalentColumns(projectedColumnNames, schema) && isEquivalentColumnTypes(projectedColumnTypes, schema);
+    }
+
+    private boolean isEquivalentColumns(Set<String> projectedColumnNames, Properties schema)
+    {
+        Set<String> columnNames;
+        String columnNameProperty = schema.getProperty(LIST_COLUMNS);
+        if (columnNameProperty.length() == 0) {
+            columnNames = new HashSet<>();
+        }
+        else {
+            final String columnNameDelimiter = (String) schema.getOrDefault(COLUMN_NAME_DELIMITER, ",");
+            columnNames = new HashSet<>(asList(columnNameProperty.split(columnNameDelimiter)));
```

Comment:
This can be simplified to `Set.of(columnNameProperty.split(columnNameDelimiter))`

Reply:
Thanks. Fixed this.